src/test/java/teammates/ui/webapi/GetNotificationActionTest.java

```diff
@@ -0,0 +1,179 @@
+
+        ______TS("unknown target user");
+        loginAsInstructor(instructor.getGoogleId());
+        requestParams = new String[] {
+                Const.ParamsNames.NOTIFICATION_TARGET_USER, "unknown",
+                Const.ParamsNames.NOTIFICATION_IS_FETCHING_ALL, String.valueOf(true),
+        };
+        verifyHttpParameterFailureAcl(requestParams);
+
+        ______TS("accessible to admin");
+        loginAsAdmin();
+        requestParams = new String[] {
+                Const.ParamsNames.NOTIFICATION_TARGET_USER, NotificationTargetUser.STUDENT.toString(),
+                Const.ParamsNames.NOTIFICATION_IS_FETCHING_ALL, String.valueOf(true),
+        };
+        verifyCanAccess(requestParams);
+    }
+
+    @Test
+    public void testExecute_withValidUserTypeForNonAdmin_shouldReturnData() {
+        ______TS("Request to fetch notification");
+        int expectedNumberOfNotifications = 4;
+        InstructorAttributes instructor = typicalBundle.instructors.get("instructor1OfCourse1");
+        loginAsInstructor(instructor.getGoogleId());
+        NotificationAttributes notification = typicalBundle.notifications.get("notification-5");
+
+        String[] requestParams = new String[] {
+                Const.ParamsNames.NOTIFICATION_TARGET_USER, NotificationTargetUser.INSTRUCTOR.toString(),
+                Const.ParamsNames.NOTIFICATION_IS_FETCHING_ALL, String.valueOf(true),
+        };
+
+        GetNotificationAction action = getAction(requestParams);
+        JsonResult jsonResult = getJsonResult(action);
+
+        NotificationsData output = (NotificationsData) jsonResult.getOutput();
+        List<NotificationData> notifications = output.getNotifications();
+
+        assertEquals(expectedNumberOfNotifications,
+                logic.getActiveNotificationsByTargetUser(notification.getTargetUser()).size());
+        assertEquals(expectedNumberOfNotifications, notifications.size());
+
+        NotificationData firstNotification = notifications.get(0);
+        assertEquals("notification-5", firstNotification.getNotificationId());
+        assertEquals(NotificationType.TIPS, firstNotification.getNotificationType());
+        assertEquals(NotificationTargetUser.INSTRUCTOR, firstNotification.getTargetUser());
+        assertEquals("The first tip to instructor", firstNotification.getTitle());
+        assertEquals("The first tip content", firstNotification.getMessage());
+    }
+
+    @Test
+    public void testExecute_withUserTypeForAdmin_shouldReturnData() {
+        ______TS("Admin request to fetch notification");
+        int expectedNumberOfNotifications = 4;
+        loginAsAdmin();
+        NotificationAttributes notification = typicalBundle.notifications.get("notification-6");
+
+        String[] requestParams = new String[] {
+                Const.ParamsNames.NOTIFICATION_TARGET_USER, NotificationTargetUser.STUDENT.toString(),
+                Const.ParamsNames.NOTIFICATION_IS_FETCHING_ALL, String.valueOf(true),
```

Comment:
```suggestion
        String[] requestParams = new String[] {
                Const.ParamsNames.NOTIFICATION_TARGET_USER, null,
                Const.ParamsNames.NOTIFICATION_IS_FETCHING_ALL, String.valueOf(true),
```

Not wrong, but for the same reason mentioned earlier.